fetch/handler.go

```diff
@@ -20,15 +20,17 @@ import (
 type handler struct {
 	logger log.Log
 	cdb    *datastore.CachedDB
+	cfg    Config
 	bs     *datastore.BlobStore
 	msh    meshProvider
 	beacon system.BeaconGetter
 }
 
-func newHandler(cdb *datastore.CachedDB, bs *datastore.BlobStore, m meshProvider, b system.BeaconGetter, lg log.Log) *handler {
+func newHandler(cdb *datastore.CachedDB, cfg Config, bs *datastore.BlobStore, m meshProvider, b system.BeaconGetter, lg log.Log) *handler {
```

Comment:
What about just passing MaxHashesInReq here, since it is the only config value used?

Reply:
`newHandler` is part of the fetcher, so I see no issue passing in the fetcher config.

Imo it shouldn't only consider `MaxHashesInReq` but also other configuration like `RequestTimeout` and `MaxRetriesForRequest`, but this is out of the scope for this PR.

Reply:
I'd say no point passing the whole object if only one field is used, but in this case if it should be using more fields from the config I think it's fine.